src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Objects/Methods.jl

```diff
@@ -7,13 +7,22 @@ function dehomogenization_map(
     X::AbsProjectiveScheme{<:Ring}, 
     U::AbsSpec
   )
+  if !isdefined(X, :dehomogenization_cache)
+    X.dehomogenization_cache = WeakKeyDict()
+  end
+  cache = X.dehomogenization_cache
+  if haskey(cache, U)
+    return cache[U]
```

Comment:
Problem here: `WeakKeyDict` is using `==` for comparison of keys. We need to wait for the `WeakKeyIdDict`.

Reply:
o.k. now we check first that it `===` one of the charts. That should be good enough since none of the standard charts 
is `==`.  Alternatively we can just make it an `IdDict`

Reply:
I think we should make it an `IdDict` for the time being and leave a TODO to change it eventually. Everything else has proved to introduce hard-to-find bugs.

Reply:
o.k. Ready for merge from my side.